lib/classes/ConfigSchemaHandler/errorMessageBuilder.js

```diff
@@ -0,0 +1,169 @@
+      if (ajvError.params && ajvError.params.additionalProperty) {
+        additionalProperty = ajvError.params.additionalProperty;
+      }
+      const formattedDataPath = getFormattedDataPath(ajvError.dataPath);
+      return `Unrecognized properety '${additionalProperty}' on '${formattedDataPath}'`;
+    }
+    default:
+      return `${getFormattedDataPath(ajvError.dataPath)} ${ajvError.message}`;
+  }
+};
+
+const addUserFriendlyMessage = (errors, userConfig) => {
+  return errors.map(error => {
+    let friendlyMessage;
+
+    // regex matches '.functions['xxx'].events[x].xxx'
+    const isEventParam = /^\.functions\['[\w-_]+'\]\.events\[[0-9]+\]\.[\w]+$/.test(error.dataPath);
+    if (error.params && error.params.additionalProperty && isEventParam) {
+      const eventName = error.dataPath.split('.')[3];
+      friendlyMessage = `Unsupported parameter '${error.params.additionalProperty}' for '${eventName}' event`;
+    }
+
+    // regex matches '.functions['xxx'].events[x]'
+    const isFunctionEvent = /^\.functions\['[\w-_]+'\]\.events\[[0-9]+\]$/.test(error.dataPath);
+    if (error.params && error.params.additionalProperty && isFunctionEvent) {
+      friendlyMessage = `Unsupported function event '${
+        error.params.additionalProperty
+      }' at ${getFormattedDataPath(error.dataPath)}`;
+    }
+
+    const isEventAnyOfErrorObject =
+      error.keyword === 'anyOf' &&
+      error.schemaPath === ANY_OF_EVENT_SCHEMA_PATH &&
+      error.params.suberrors &&
+      error.params.suberrors.length &&
+      error.params.suberrors[0].params &&
+      error.params.suberrors[0].params.additionalProperty;
+
+    if (isEventAnyOfErrorObject) {
+      friendlyMessage = `Unsupported function event '${
+        error.params.suberrors[0].params.additionalProperty
+      }' at ${getFormattedDataPath(error.dataPath)}`;
+    }
+
+    const configUnderValidation = _.get(userConfig, error.dataPath.slice(1));
+
+    if (isEventAnyOfErrorObject && Object.keys(configUnderValidation).length > 1) {
+      const contextStr = `${Object.keys(configUnderValidation).length} (${Object.keys(
+        configUnderValidation
+      ).join(', ')})`;
+      friendlyMessage = [
+        'Event should contain only one root property,',
+        `but got ${contextStr} at ${getFormattedDataPath(error.dataPath)}`,
+      ].join(' ');
+    }
+
+    if (error.dataPath === '.functions' && error.params && error.params.additionalProperty) {
+      friendlyMessage = `Function name '${
+        error.params.additionalProperty
+      }' must be alphanumeric at ${getFormattedDataPath(error.dataPath)}.`;
+    }
+
+    if (friendlyMessage) {
+      error.friendlyMessage = friendlyMessage;
+    }
+
+    return error;
+  });
+};
+
+function getFormattedDataPath(dataPath) {
+  const result = dataPath === '' ? 'root' : dataPath.slice(1);
+
+  // This regex helps replace functions['someFunc'].foo with functions.someFunc.foo
+  const bracketsRegex = /\['(.+)'\]/g;
```

Comment:
I would replace just names that work as written literally, so `foo['bar']` should be replaced with `foo.bar`, but `foo['b-a-r']` should stay in original form imo

Reply:
I agree. Adjusted in https://github.com/serverless/serverless/pull/7335/commits/4abac84afd74835efced7a75baaafe68d227e32f